Failure renaming kubernetes cloud-provider

Attempt to change the name of cloud provider fails. This causes

2021/10/11 13:38:57 [ERROR] eval: *terraform.EvalRefresh, err: received http status code '400'
2021/10/11 13:38:57 [ERROR] eval: *terraform.EvalSequence, err: received http status code '400'

Error: received http status code '400'
to be returned by terraform. The state file does get updated however and has the right name.

After renaming provider manually in UI as well as selecting associated Service Account Token works and fixes the above error.

--- a/internal/provider/cloudprovider.go
+++ b/internal/provider/cloudprovider.go
@@ -19,6 +19,7 @@ func commonCloudProviderSchema() map[string]*schema.Schema {
 			Description: "The name of the cloud provider.",
 			Type:        schema.TypeString,
 			Required:    true,
+			ForceNew:    true,
 		},
 		"usage_scope": usageScopeSchema(),
 	}

--- a/internal/provider/resource_cloudprovider_kubernetes_test.go
+++ b/internal/provider/resource_cloudprovider_kubernetes_test.go
@@ -16,6 +16,7 @@ func TestAccResourceK8sCloudProviderConnector_delegate(t *testing.T) {
 
 	var (
 		name         = fmt.Sprintf("%s_%s", t.Name(), utils.RandStringBytes(4))
+		updatedName  = fmt.Sprintf("%s_updated", name)
 		resourceName = "harness_cloudprovider_kubernetes.test"
 	)
 
@@ -29,6 +30,13 @@ func TestAccResourceK8sCloudProviderConnector_delegate(t *testing.T) {
 				Check: resource.ComposeTestCheckFunc(
 					resource.TestCheckResourceAttr(resourceName, "name", name),
 					testAccCheckK8sCloudProviderExists(t, resourceName, name),
+				),
+			},
+			{
+				Config: testAccResourceK8sCloudProvider_delegate(updatedName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "name", updatedName),
+					testAccCheckK8sCloudProviderExists(t, resourceName, updatedName),
 				),
 			},
 			{

